ovirt_sriov.py

```diff
@@ -0,0 +1,384 @@
+
+        return nic_with_attributes[0].virtual_functions_configuration
+
+    def get_vf_network_ids(self, nic_service):
+        networks_list = nic_service.virtual_function_allowed_networks_service().list()
+        network_ids = []
+        if networks_list:
+            for network in networks_list:
+                network_ids.append(network.id)
+
+        return network_ids
+
+    def get_vf_labels(self, nic_service):
+        vf_labels = nic_service.virtual_function_allowed_labels_service().list()
+        labels = []
+        if vf_labels:
+           for label in vf_labels:
+              labels.append(label.id)
+
+        return labels
+
+    def has_updates(self, networks_service, nics_service, nic_service):
+        update = False
+        allowed_networks = self._module.params['allowed_networks']
+        vfs = self._module.params['vfs']
+        vf_labels = self._module.params['labels']
+        vf_config = self.get_vf_config(nics_service)
+
+        # Check if allowed_networks needs updating
+        if allowed_networks:
+            all_networks = True if allowed_networks == 'all' else False
+
+            if all_networks != vf_config.all_networks_allowed:
+                update = True
+
+            # Check if specific networks need updating
+            if not all_networks:
+                existing_networks = self.get_vf_network_ids(nic_service)
+                desired_networks = self.get_network_ids(networks_service)
+                if sorted(existing_networks) != sorted(desired_networks):
+                    update = True
+
+        # Check if labels need updating
+        if vf_labels and allowed_networks == 'specific':
+            vf_existing_labels = self.get_vf_labels(nic_service)
+            if sorted(vf_labels) != sorted(vf_existing_labels):
+                update = True
+
+        # Check if the number of vfs needs updating
+        if vfs:
+            if vfs != vf_config.number_of_virtual_functions:
+                update = True
+
+        # Finally, return result
+        return update
+
+    def update_vf_labels(self, nic_service):
+        vf_labels = self._module.params['labels']
+        vf_labels_service = nic_service.virtual_function_allowed_labels_service()
+        vf_existing_labels = self.get_vf_labels(nic_service)
+        changed = False
+
+        # Remove any unwanted labels
+        remove_labels = list(set(vf_existing_labels).difference(vf_labels))
+        if remove_labels:
+            for label in remove_labels:
+                vf_labels_service.service(label).remove()
+            changed = True
+
+        # Add any new labels
+        add_labels = list(set(vf_labels).difference(vf_existing_labels))
+        if add_labels:
+            for label in add_labels:
+                vf_labels_service.add(otypes.NetworkLabel(
+                    id=label,
+                    host_nic=nic_service.get()
+                )

```

Comment:
It's more of a style thing, but I would put both of these on the same line since they are closing out something that is in the same level of code block.

Reply:
Good catch, I'll change it.